java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -245,6 +245,9 @@ protected final void serializeSomethingField(
             StructBondType.StructField<T> field) throws IOException {
         if (value != null) {
             serializeField(context, value.getValue(), field);
+        } else if (!field.isOptional()) {
```

Comment:
Yes, it's used to serialize every field that defaults to nothing and is not a Java primitive type.

---
In reply to: [134046697](https://github.com/Microsoft/bond/pull/572#discussion_r134046697) [](ancestors = 134046697)